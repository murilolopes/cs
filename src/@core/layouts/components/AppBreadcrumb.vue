<template>
  <b-row
    v-if="$route.meta.breadcrumb || $route.meta.pageTitle"
    class="content-header px-2"
  >
    <!-- Content Left -->
    <b-col class="content-header-left mb-2" cols="12" md="9">
      <b-row class="breadcrumbs-top">
        <b-col cols="12" class="d-flex align-items-center">
          <h1
            class="float-left pr-1 mb-0"
            :class="{ 'content-header-title': !$route.meta.skipGoToHome }"
          >
            {{ $route.meta.pageTitle }}
          </h1>
          <div class="breadcrumb-wrapper" v-if="!$route.meta.skipGoToHome">
            <b-breadcrumb>
              <b-breadcrumb-item to="/">
                <feather-icon
                  icon="HomeIcon"
                  size="16"
                  class="align-text-top"
                />
              </b-breadcrumb-item>
              <b-breadcrumb-item
                v-for="item in $route.meta.breadcrumb"
                :key="item.text"
                :active="item.active"
                :to="item.to"
              >
                {{ item.text }}
              </b-breadcrumb-item>
            </b-breadcrumb>
          </div>
        </b-col>
        <b-col
          cols="12"
          class="d-flex align-items-center"
          v-if="$route.meta.pageSubtitle"
        >
          <h4 class="float-left pr-1 mt-1 mb-0">
            {{ $route.meta.pageSubtitle }}
          </h4>
        </b-col>
      </b-row>
    </b-col>

    <!-- Content Right -->
    <b-col
      v-if="$route.meta.actions?.length"
      class="content-header-right text-md-right d-md-block d-none mb-1"
      md="3"
      cols="12"
    >
      <b-button
        v-for="action in $route.meta.actions"
        :key="action.title"
        :variant="action.variant"
        :to="{ name: action.routeName }"
      >
        <feather-icon :icon="action.icon" class="mr-50" v-if="action.icon" />
        <span class="align-middle">{{ action.title }}</span>
      </b-button>
    </b-col>
  </b-row>
</template>

<script>
import {
  BBreadcrumb,
  BBreadcrumbItem,
  BRow,
  BCol,
  BDropdown,
  BDropdownItem,
  BButton,
} from "bootstrap-vue";
import Ripple from "vue-ripple-directive";

export default {
  directives: {
    Ripple,
  },
  components: {
    BBreadcrumb,
    BBreadcrumbItem,
    BRow,
    BCol,
    BDropdown,
    BDropdownItem,
    BButton,
  },
};
</script>
