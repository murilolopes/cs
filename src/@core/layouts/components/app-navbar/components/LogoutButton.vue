<template>
  <div class="d-flex align-items-center justify-content-center cursor-pointer" @click="logout()">
    <h3 class="mb-0 mr-1">Sair</h3>
     <feather-icon icon="LogInIcon" size="24" />
  </div>
</template>

<script>

export default {
  methods: {
    logout() {
      this.$store.dispatch('auth/logout')
    },
  },
}
</script>
