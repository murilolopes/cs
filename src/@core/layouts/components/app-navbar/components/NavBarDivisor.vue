<template>
  <div class="nav-bar-divisor" />
</template>

<style>
  .nav-bar-divisor {
    height: 3rem;
    margin-left: 2rem;
    margin-right: 2rem;
    border-left: 1px solid #DAE1E8;
  }
</style>