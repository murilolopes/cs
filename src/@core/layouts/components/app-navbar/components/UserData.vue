 <template>
  <div class="d-flex align-items-center cursor-pointer" @click="myAccount">
    <div class="d-flex flex-column align-items-end mr-1">
      <h6 class="mb-0">{{ $store.state.auth.userData.name }}</h6>
      <span>{{ userSubTitle() }}</span>
    </div>
    <feather-icon icon="UserIcon" size="24" />
  </div>
</template>

<script>
export default {
  name: 'UserData',
  methods: {
    myAccount() {
      const userAccountRoute = {admin: 'admin.minha-conta', cedente: 'minha-conta', parceiro: 'parceiro.minha-conta'}
      this.$router.push({ name: userAccountRoute[this.userType] })
    },
    userSubTitle() {
      return this.userType === 'admin' ? 'Administrador' : `${this.$store.state.auth.currentEconomicGroup?.nome}`
    }
  },
  computed: {
    formattedCnpj() {
      return this.$store.state.auth.userData.empresa.cnpj.replace(/^(\d{2})(\d{3})(\d{3})(\d{4})(\d{2})/, "$1.$2.$3/$4-$5")
    },
    userType() {
      return this.$store.state.auth.userData.user_type
    }
  }
}
</script>
