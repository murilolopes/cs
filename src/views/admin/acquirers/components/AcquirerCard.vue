<template>
  <b-card class="d-flex flex-row align-items-center cc-bg-light py-1 px-2" no-body>
    <div class="d-flex flex-row align-items-center p-0 border-right">
      <b-img :src="acquirerData.logo"
             class="mr-3"
             fluid
             width="112.28px"
             height="64px"
      />
    </div>
    <div class="d-flex flex-row justify-content-between align-items-center p-0 ml-2 w-100">
      <div class="d-flex flex-row align-items-center p-0">
        <div class="d-flex flex-column align-items-center p-0 mr-3">
          <div class="d-flex flex-column align-items-start p-0">
            <span :class="{ 'disabled': !acquirerData.enable }">Nome Fantasia</span>
            <h6 :class="{ 'disabled': !acquirerData.enable }">
              {{ acquirerData.nome_fantasia }}
            </h6>
          </div>
        </div>
        <div class="d-flex flex-column align-items-center p-0">
          <div class="d-flex flex-column align-items-start p-0">
            <span :class="{ 'disabled': !acquirerData.enable }">CNPJ</span>
            <h6 :class="{ 'disabled': !acquirerData.enable }">
              {{ acquirerData.cnpj | cnpjFormatter }}
            </h6>
          </div>
        </div>
        <div class="d-flex flex-row align-items-end p-0 ml-3">
          <div class="d-flex flex-column align-items-center p-0">
            <div class="d-flex flex-column align-items-start p-0">
              <span :class="{ 'disabled': !acquirerData.enable }">Taxa global</span>
              <h6 :class="{ 'disabled': !acquirerData.enable }">
                {{ updatedTaxaJurosPadrao | percentageFormatter }}
              </h6>
            </div>
          </div>
          <b-button variant="link" @click="showTaxesTimeline">
            <feather-icon icon="LayersIcon" size="20" />
          </b-button>
          <b-button variant="link" :disabled="!acquirerData.enable" @click="showTaxEdition">
            <feather-icon icon="Edit3Icon" size="20" />
          </b-button>
        </div>
      </div>
      <div class="d-flex flex-row justify-content-end align-items-center p-0">
        <div class="d-flex flex-row align-items-start p-0 mr-2">
          <b-form-checkbox v-model="acquirerData.enable"
                           class="custom-control-success"
                           name="enable-button"
                           switch
                           @change="updateState"
          >
            <span class="switch-icon-left"><feather-icon icon="CheckIcon" /></span>
            <span class="switch-icon-right"><feather-icon icon="XIcon" /></span>
          </b-form-checkbox>
          Apto para operar
        </div>
        <b-button v-if="editable"
                  variant="outline-primary"
                  size="sm"
                  @click="$router.push({ name: 'admin-credenciadoras.show', params: { id: acquirerData.id } })"
        >
          Editar
        </b-button>
      </div>
    </div>
    <taxes-timeline-modal ref="taxesTimelineModal"
                          :chart-data="computedSeries"
                          :chart-options="options"
                          :logo="acquirerData.logo"
    />
    <tax-edition-modal ref="taxEditionModal"
                       :current-acquirer-id="acquirerData.id"
                       :current-tax="acquirerData.taxa_juros_padrao"
                       @update="handleGlobalTaxUpdate"
    />
    <change-enable-modal ref="changeEnableModal"
                         :current-acquirer-id="acquirerData.id"
                         :current-enable-state="acquirerData.enable"
                         @cancel="handleEnable"
    />
  </b-card>
</template>

<script>
import {
  BCard, BImg, BButton, BFormCheckbox,
} from 'bootstrap-vue'
import { cnpj as cnpjFormatter, percentageFormatter } from '@core/comp-functions/data_visualization/datatable'
import TaxesTimelineModal from '@views/admin/acquirers/components/modals/TaxesTimelineModal.vue'
import TaxEditionModal from '@views/admin/acquirers/components/modals/TaxEditionModal.vue'
import ChangeEnableModal from '@views/admin/acquirers/components/modals/ChangeEnableModal.vue'

export default {
  name: 'AcquirerCard',
  components: {
    BCard,
    BImg,
    BButton,
    BFormCheckbox,
    TaxesTimelineModal,
    TaxEditionModal,
    ChangeEnableModal,
  },
  filters: { cnpjFormatter, percentageFormatter },
  props: { acquirerData: { type: Object, default: () => { } }, editable: Boolean },
  data() {
    return {
      options: {
        chart: {
          id: 'taxes-timeline',
          toolbar: {
            show: false,
          },
        },
        legend: {
          position: 'top',
          verticalAlign: 'top',
          horizontalAlign: 'start',
          showForSingleSeries: true,
          containerMargin: {
            left: 35,
            right: 60,
          },
        },
        markers: {
          size: 0,
        },
        stroke: {
          width: 5,
          curve: 'smooth',
        },
        grid: {
          row: {
            colors: ['transparent', 'transparent'],
            opacity: 0.5,
          },
        },
        colors: ['#3FE4A9'],
        yaxis: {
          labels: {
            formatter(value) {
              return `${value.toFixed(1)}%`
            },
          },
        },
        xaxis: {
          type: 'datetime',
          labels: {
            format: 'dd/MM',
          },
        },
      },
      taxesData: [],
      taxaJurosPadrao: this.acquirerData.taxa_juros_padrao,
    }
  },
  computed: {
    computedSeries() {
      const taxesByDate = []
      this.taxesData.map(item => taxesByDate.push({ x: item.date, y: JSON.parse(item.taxa_juros_padrao) }))
      return [{
        name: 'Taxa Global',
        data: taxesByDate,
      }]
    },
    updatedTaxaJurosPadrao() {
      return this.taxaJurosPadrao
    },
  },
  methods: {
    async showTaxesTimeline() {
      await this.$refs.taxesTimelineModal.$refs.taxesTimelineModal.show()
      const { data } = await this.$store.dispatch('credenciadora/fetchTaxesHistory', this.acquirerData.id)
      this.taxesData = data
    },
    showTaxEdition() {
      this.$refs.taxEditionModal.$refs.taxEditionModal.show()
    },
    handleGlobalTaxUpdate(taxaJurosPadrao) {
      this.taxaJurosPadrao = taxaJurosPadrao
    },
    handleEnable(currentEnableState) {
      this.acquirerData.enable = currentEnableState
    },
    updateState() {
      this.$refs.changeEnableModal.$refs.changeEnableModal.show()
    },
  },
}

</script>

<style lang="scss">

.disabled {
  color: #B4C0CC;
}

</style>
