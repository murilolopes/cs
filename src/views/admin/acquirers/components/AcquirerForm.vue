<template>
  <div>
    <b-form>
      <validation-observer ref="AcquirerForm" #default="{ invalid }">
        <b-form-group label="Dados da Credenciadora" label-class="font-weight-bold" label-cols="12">
          <b-form-row>
            <b-col cols="3">
              <b-form-group label="Nome Fantasia" label-for="nome_fantasia">
                <validation-provider #default="{ errors }"
                                     name="nome_fantasia"
                                     vid="nome_fantasia"
                                     rules="required"
                >
                  <b-form-input id="nome_fantasia"
                                v-model="credenciadora.nome_fantasia"
                                :state="errors.length > 0 ? false : null"
                                name="nome_fantasia"
                  />
                  <small class="text-danger">{{ errors[0] }}</small>
                </validation-provider>
              </b-form-group>
            </b-col>
            <b-col cols="3">
              <b-form-group label="CNPJ" label-for="cnpj">
                <validation-provider #default="{ errors }" name="cnpj" vid="cnpj" rules="required|cnpj">
                  <b-form-input id="cnpj"
                                v-model="credenciadora.cnpj"
                                v-mask="['##.###.###/####-##']"
                                :state="errors.length > 0 ? false : null"
                                name="cnpj"
                  />
                  <small class="text-danger">{{ errors[0] }}</small>
                </validation-provider>
              </b-form-group>
            </b-col>
            <b-col cols="3">
              <b-form-group label="Taxa Global" label-for="taxa_juros_global">
                <validation-provider #default="{ errors }"
                                     name="taxa_juros_global"
                                     vid="taxa_juros_global"
                                     rules="required"
                >
                  <b-form-input id="taxa_juros_global"
                                v-model="credenciadora.taxa_juros_padrao"
                                v-mask="'#.##'"
                                :state="errors.length > 0 ? false : null"
                                name="taxa_juros_global"
                  />
                  <small class="text-danger">{{ errors[0] }}</small>
                </validation-provider>
              </b-form-group>
            </b-col>
            <b-col cols="3" align-self="center">
              <b-form-group v-slot="{ ariaDescribedby }"
                            label="Apto para operar"
                            label-for="apto_para_operar"
              >
                <validation-provider #default="{ errors }"
                                     name="apto_para_operar"
                                     vid="apto_para_operar"
                                     rules="required"
                >
                  <b-form-checkbox id="apto_para_operar"
                                   v-model="credenciadora.enable"
                                   class="custom-control-success"
                                   name="apto_para_operar"
                                   :aria-describedby="ariaDescribedby"
                                   switch
                  >
                    <span class="switch-icon-left"><feather-icon icon="CheckIcon" /></span>
                    <span class="switch-icon-right"><feather-icon icon="XIcon" /></span>
                  </b-form-checkbox>
                  <small class="text-danger">{{ errors[0] }}</small>
                </validation-provider>
              </b-form-group>
            </b-col>
          </b-form-row>
          <b-form-row>
            <b-col cols="3">
              <b-form-group label="Logo da Credenciadora" label-class="font-weight-bold" label-for="logo" label-cols="12">
                <validation-provider #default="{ errors }"
                name="logo"
                vid="logo"
                rules="required"
                >
                <b-form-input id="logo"
                v-model="credenciadora.logo"
                :state="errors.length > 0 ? false : null"
                name="logo"
                  />
                  <small class="text-danger">{{ errors[0] }}</small><br>
                  <small class="image-title">Pré-visualização:</small><br>
                </validation-provider>
                <b-img
                  v-if="credenciadora.logo"
                  height="80px"
                  width="140px"
                  class="mt-2 ml-2"
                  :src="credenciadora.logo"
                  fluid
                  alt="Logo da Credenciadora"
                />
              </b-form-group>
            </b-col>
          </b-form-row>
        </b-form-group>
        <b-form-row class="mb-1">
          <b-col cols="12" class="d-flex flex-row justify-content-end">
            <b-button variant="outline-primary" class="mr-1" @click="$emit('cancel')">
              Cancelar
            </b-button>
            <b-button variant="primary" :disabled="invalid" @click="save">
              Salvar credenciadora
            </b-button>
          </b-col>
        </b-form-row>
      </validation-observer>
    </b-form>
  </div>
</template>

<script>
import {
  BForm, BFormGroup, BFormCheckbox, BFormRow, BCol, BButton, BFormInput, BImg
} from 'bootstrap-vue'
import { ValidationProvider, ValidationObserver } from 'vee-validate'
import { required, cnpj } from '@validations'
import { mask } from 'vue-the-mask'
import CcDropzone from '@core/components/cc-dropzone/CcDropzone.vue'

export default {
  name: 'AcquirerForm',
  components: {
    BForm,
    BFormGroup,
    BFormRow,
    BFormInput,
    BFormCheckbox,
    BCol,
    BButton,
    BImg,
    ValidationProvider,
    ValidationObserver,
    CcDropzone,
  },
  directives: { mask },
  data() {
    return {
      credenciadora: {
        nome_fantasia: '',
        razao_social: '',
        cnpj: '',
        enable: true,
        taxa_juros_padrao: '',
        logo: '',
      },
      required,
      cnpj,
      isEditing: false,
    }
  },
  async mounted() { if (this.$route.params.id) await this.getData() },
  methods: {
    async getData() {
      const { data } = await this.$store.dispatch('credenciadora/fetch', this.$route.params.id)
      this.credenciadora = data
      this.isEditing = true
    },
    async save() {
      try {
        this.$swal.showLoading()
        await this.$store.dispatch('credenciadora/save', this.credenciadora)
        await this.$swal.fire({
          title: 'Sucesso!',
          text: 'Dados cadastrados com sucesso!',
          icon: 'success',
          confirmButtonText: 'Ok',
        })
        this.$router.push({
          name: 'admin-credenciadoras.index',
        })
      } catch (error) {
        this.$swal.fire({
          title: 'Erro!',
          text: 'Não foi possível completar a operação!',
          icon: 'error',
          confirmButtonText: 'Ok',
        })
      }
    },
  },
}
</script>

<style scoped>
</style>
