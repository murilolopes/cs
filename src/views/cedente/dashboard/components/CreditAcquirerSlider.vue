<template>
  <b-row>
    <b-col sm="12">
      <div class="d-flex flex-row align-items-center pb-2 mb-2">
        <div class="d-flex flex-column justify-content-center align-items-center w-25 text-center">
          <b-img
            height="36px"
            width="104px"
            :src="acquirer.logo"
            alt="Logo da credenciadora"
          />
          <p>{{ acquirer.name }}</p>
          <h3>Taxa de {{ acquirer.tax }}%</h3>
        </div>
        <div class="d-flex flex-column align-items-start p-0 w-75">
          <div v-if="!disabledSlider" class="d-flex flex-row justify-content-between align-items-start p-0 w-100">
            <p class="mb-0">
              Selecionado
            </p>
            <p class="mb-0">
              Total disponível
            </p>
          </div>
          <div v-else class="d-flex flex-row justify-content-between align-items-center w-100">
            <p class="mb-0">
              {{ disabledMessage }}
            </p>
            <b-button v-if="successWebhook" variant="outline-primary" size="sm" @click="goToSolicitation">
              Saiba mais
            </b-button>
          </div>
          <vue-slider
            v-model="acquirer.selectedValue"
            :min="0"
            :tooltip-formatter="(val) => moneyFormat(val, 0)"
            :max="acquirer.totalAmountValue"
            :interval="1000"
            class="w-100 mt-1 mb-1 cc-slider"
            :lazy="true"
            :disabled="disabledSlider"
          />
          <div class="d-flex flex-row justify-content-between align-items-start w-100 mb-0">
            <div class="d-flex align-items-center">
              <validation-provider
                #default="{ errors }"
                name="selectedValue"
                vid="selectedValueInput"
                rules="multipleOf1000"
              >
                <b-form-input
                  id="selectedValueInput"
                  v-model="acquirer.selectedValueInput"
                  :disabled="disabledSlider"
                  class="w-100 text-dark"
                  :state="errors.length > 0 ? false : null"
                  name="selectedValueInput"
                  @blur="updateSelectedValueInput"
                />
                <small class="text-danger">{{ errors[0] }}</small>
              </validation-provider>

              <b-badge v-if="acquirer.effective_tax > 0" variant="light-warning" class="ml-1 border-warning text-black">
                Taxa efetiva {{ effectiveTax }}
              </b-badge>
            </div>
            <validation-provider
              #default="{ errors }"
              name="totalAmountValue"
              vid="totalAmountValue"
              rules="multipleOf1000"
            >
              <b-form-input
                id="totalAmountValueInput"
                v-model="acquirer.totalAmountValueInput"
                class="w-100"
                :state="errors.length > 0 ? false : null"
                name="totalAmountValueInput"
                disabled
              />
              <small class="text-danger">{{ errors[0] }}</small>
            </validation-provider>
          </div>
        </div>
      </div>
    </b-col>

    <b-col sm="12" class="pl-3 pr-2 mt-0 mb-1">
      <b-card-actions title="Valor acumulado no período" action-collapse collapsed class="card-actions-border">
        <vue-apex-charts
          v-if="cumulativeAmountChartSeries[0].data.length > 0"
          height="200"
          :options="urs.chartOptions"
          :series="cumulativeAmountChartSeries"
        />
      </b-card-actions>
    </b-col>
  </b-row>
</template>

<script>
import {
  BRow, BCol, BImg, BFormInput, BButton, BBadge,
} from 'bootstrap-vue'
import VueSlider from 'vue-slider-component'
import { moneyFormat, percentageFormatter } from '@core/comp-functions/data_visualization/datatable'
import { brlToFloat } from '@core/comp-functions/money/money'
import { ValidationProvider } from 'vee-validate'
import { multiple1000 } from '@validations'
import { BCardActions } from '@/@core/components/b-card-actions'
import VueApexCharts from 'vue-apexcharts'

export default {
  name: 'CreditAcquirerSlider',
  components: {
    BCardActions,
    BBadge,
    BRow,
    BCol,
    BImg,
    BButton,
    VueSlider,
    BFormInput,
    ValidationProvider,
    VueApexCharts,
  },
  props: {
    acquirerData: {
      type: Object,
      default: () => {},
    },
  },
  data() {
    return {
      acquirer: null,
      multiple1000,
      urs: {
        chartOptions: {
          chart: {
            toolbar: {
              show: false,
            },
          },
          dataLabels: {
            enabled: false,
          },
          stroke: {
            show: true,
            curve: 'straight',
          },
          legend: {
            show: true,
            position: 'top',
            horizontalAlign: 'left',
            fontSize: '14px',
            fontFamily: 'Montserrat',
          },
          grid: {
            xaxis: {
              lines: {
                show: true,
              },
            },
          },
          yaxis: {
            labels: {
              formatter(value) {
                return moneyFormat(value)
              },
            },
          },
          xaxis: {
            type: 'datetime',
            labels: {
              format: 'dd/MM',
            },
          },
          tooltip: {
            shared: false,
          },
          colors: ['#3FE4A9'],
        },
      },
    }
  },
  computed: {
    effectiveTax() {
      return percentageFormatter(parseFloat(this.acquirer.effective_tax))
    },
    successWebhook() {
      return this.acquirer.inProgressSolicitation?.status === 'sucesso_webhook'
    },
    disabledSlider() {
      return ['sucesso_webhook', 'confirmado', 'solicitada', 'enviada_cerc'].includes(this.acquirer.inProgressSolicitation?.status)
    },
    disabledMessage() {
      const message = {
        solicitada: 'Aguarde, solicitação em andamento.',
        confirmado: 'Já foi feita uma solicitação para a credenciadora hoje',
        enviada_cerc: 'Aguarde, solicitação em andamento.',
        sucesso_webhook: 'Já existe uma solicitação em andamento',
      }
      return message[this.acquirer.inProgressSolicitation?.status]
    },
    cumulativeAmountChartSeries() {
      const seriesData = []
      this.acquirerData.cumulativeAmount.map(item => seriesData.push(
        { x: item.data_vencimento, y: JSON.parse(item.valor_acumulado) },
      ))
      return [{
        name: 'Valor Acumulado',
        data: seriesData,
      }]
    },

  },
  watch: {
    // eslint-disable-next-line no-unused-vars, func-names
    'acquirer.selectedValue': function (newValue, oldValue) {
      if (newValue > this.acquirer.totalAmountValue) this.acquirer.selectedValueInput = moneyFormat(this.acquirer.totalAmountValue, 0)
      this.acquirer.selectedValueInput = moneyFormat(newValue, 0)
      // this.getEffectiveTax(newValue);
      this.$emit('selected-from-slider')
    },
  },
  created() {
    this.acquirer = this.acquirerData
  },
  methods: {
    refreshComponent() {
      this.acquirer = this.acquirerData
    },
    goToSolicitation() {
      this.$router.push({ name: 'antecipacoes.details', params: { id: this.acquirer.inProgressSolicitation.id } })
    },
    moneyFormat(value, minimumFractionDigits = 2) {
      return moneyFormat(parseInt(value, 10), minimumFractionDigits)
    },
    updateSelectedValueInput($event) {
      let { value } = $event.target
      if (typeof value === 'string') {
        if (brlToFloat(value) > this.acquirer.totalAmountValue) value = moneyFormat(this.acquirer.totalAmountValue, 0)
        this.acquirer.selectedValue = brlToFloat(value)
      }
      this.$emit('selected-from-input')
      this.$emit('blur', value)
    },
    async getEffectiveTax(newValue) {
      if (newValue === 0) return

      const payload = {
        credenciadora_id: this.acquirer.id,
        valor_solicitacao: newValue,
      }
      const { data } = await this.$store.dispatch('dashboard/getEffectiveTax', payload)

      this.$emit('effective-tax', { credenciadora_id: this.acquirer.id, effective_tax: data.taxa_efetiva_prevista })
    },
  },
}
</script>

<style lang="scss">
.cc-slider {
  height: 8px!important;
}

.vue-slider-dot-handle {
  background-color: #3FE4A9!important;
}

.card-actions-border {
  border: 1px solid #c4ced6;
  border-radius: 6px;

  .card-title {
    display: flex;
    align-items: center;
  }

  .card-title::before {
    content: " ";
    margin-left: 1rem;
    margin-right: 1rem;
    display: flex;
    width: 1rem;
    height: 1rem;
    background: #3FE4A9;
    border-radius: 50%;
  }

  .logo-image {
  height: 36px;
  width: 104px;
  object-fit: cover;
}
}
</style>
