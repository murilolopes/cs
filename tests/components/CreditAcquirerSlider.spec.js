import { mount, shallowMount } from '@vue/test-utils'
import CreditAcquirerSlider from '@/views/cedente/dashboard/components/CreditAcquirerSlider.vue'
import { BBadge } from 'bootstrap-vue';
import { BCardActions } from '@/@core/components/b-card-actions';
import VueApexCharts from 'vue-apexcharts';

describe('CreditAcquirerSlider.vue', () => {
  it("should render correctly", () => {
    const acquirerData = {
      tax: 1.85,
      logo: 'x'
    }

    const wrapper = shallowMount(CreditAcquirerSlider, { propsData: { acquirerData }})
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Must show effective tax badge', () => {
    const acquirerData = {
      tax: 1.85,
      logo: 'z'
    }

    const wrapper = shallowMount(CreditAcquirerSlider, { propsData: { acquirerData }})

    const efectiveTaxBadge = wrapper.findComponent(BBadge)

    expect(efectiveTaxBadge.isVisible()).toBe(true)
    expect(efectiveTaxBadge.text()).toEqual('Taxa efetiva 1,85%')
  })

  it('Must not show effective tax badge', () => {
    const acquirerData = {
      logo: 'z'
    }

    const wrapper = shallowMount(CreditAcquirerSlider, { propsData: { acquirerData }})

    const efectiveTaxBadge = wrapper.findComponent(BBadge)

    expect(efectiveTaxBadge.isVisible()).toBe(false)
  })

  it('Must show a b-card-actions collapsed by default', () => {
    const acquirerData = {
      logo: 'z'
    }

    const wrapper = shallowMount(CreditAcquirerSlider, { propsData: { acquirerData }})

    const efectiveTaxBadge = wrapper.findComponent(BCardActions)

    expect(efectiveTaxBadge.props().collapsed).toBe(true)
  })
})
