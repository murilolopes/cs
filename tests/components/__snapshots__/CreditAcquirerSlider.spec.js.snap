// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditAcquirerSlider.vue should render correctly 1`] = `
<div>
  <div
    class="d-flex flex-row align-items-center p-0 mb-2"
  >
    <div
      class="d-flex flex-column justify-content-center align-items-center w-25 text-center"
    >
      <b-img-stub
        alt="Logo da credenciadora"
        blankcolor="transparent"
        height="36px"
        src="x"
        width="104px"
      />
       
      <p>
        
      </p>
       
      <h3>
        Taxa de 1.85%
      </h3>
    </div>
     
    <div
      class="d-flex flex-column align-items-start p-0 w-75"
    >
      <div
        class="d-flex flex-row justify-content-between align-items-start p-0 w-100"
      >
        <p
          class="mb-0"
        >
          Selecionado
        </p>
         
        <p
          class="mb-0"
        >
          Total disponível
        </p>
      </div>
       
      <vue-slider-stub
        class="w-100 mt-1 mb-1 cc-slider"
        clickable="true"
        datalabel="label"
        datavalue="value"
        direction="ltr"
        dotsize="14"
        duration="0.5"
        enablecross="true"
        interval="0.01"
        lazy="true"
        max="100"
        min="0"
        order="true"
        process="true"
        tooltip="active"
        tooltipformatter="[Function]"
        usekeyboard="true"
        value="0"
      />
       
      <div
        class="d-flex flex-row justify-content-between align-items-start w-100 mb-0"
      >
        <div
          class="d-flex align-items-center"
        >
          <validation-provider-stub
            bails="true"
            custommessages="[object Object]"
            debounce="0"
            detectinput="true"
            mode="aggressive"
            name="selectedValue"
            rules="multipleOf1000"
            skipifempty="true"
            tag="span"
            vid="selectedValue"
          />
           
          <b-badge-stub
            class="ml-1 border-warning text-black"
            tag="span"
            target="_self"
            variant="light-warning"
          >
            Taxa efetiva 1,85%
          </b-badge-stub>
        </div>
         
        <validation-provider-stub
          bails="true"
          custommessages="[object Object]"
          debounce="0"
          detectinput="true"
          mode="aggressive"
          name="totalAmountValue"
          rules="multipleOf1000"
          skipifempty="true"
          tag="span"
          vid="totalAmountValue"
        />
      </div>
    </div>
  </div>
   
  <b-card-actions-stub
    actioncollapse="true"
    class="card-actions-border"
    collapsed="true"
    title="Valor acumulado no período"
  >
    <vue-apex-charts-stub
      height="200"
      options="[object Object]"
      series="[object Object]"
      width="100%"
    />
  </b-card-actions-stub>
</div>
`;
